src/main/kotlin/graphql/kickstart/tools/util/Utils.kt

```diff
@@ -49,10 +50,16 @@ internal val Class<*>.declaredNonProxyMethods: List<JavaMethod>
         }
     }
 
-internal fun getDocumentation(node: AbstractNode<*>): String? = node.comments?.asSequence()
-    ?.filter { !it.content.startsWith("#") }
-    ?.joinToString("\n") { it.content.trimEnd() }
-    ?.trimIndent()
+internal fun getDocumentation(node: AbstractDescribedNode<*>, options: SchemaParserOptions): String? =
+    when {
+        node.description != null -> node.description.content
+        !options.useCommentsForDescriptions -> null
+        node.comments.isNullOrEmpty() -> null
```

Comment:
These two conditions both return `null` so you could just combine them into one.